modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -140,6 +141,34 @@ class Params<cv::gapi::Generic> {
     std::string m_tag;
 };
 
+// NB: Used by python wrapper
+// This class can be marked as SIMPLE, because it's implemented as pimpl
+class GAPI_EXPORTS_W_SIMPLE PyParams {
+public:
+    PyParams() = default;
+
+    PyParams(const std::string &tag,
+             const std::string &model,
+             const std::string &weights,
+             const std::string &device)
+        : m_priv(std::make_shared<Params<cv::gapi::Generic>>(tag, model, weights, device)) {
+    }
+
+    GBackend      backend()    const { return m_priv->backend(); }
+    std::string   tag()        const { return m_priv->tag();     }
+    cv::util::any params()     const { return m_priv->params();  }
+
+private:
+    std::shared_ptr<Params<cv::gapi::Generic>> m_priv;
+};
```

Comment:
Can we just define a shadow class Params?

Reply:
Doesn't work on `clang`